### PR TITLE
[formrecognizer] Improve live pipeline tests

### PR DESCRIPTION
**Pull request by catalinaperalta**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/23227
Add a sleep after each test to improve pipeline health meanwhile the service team looks into the flaky errors related to accessing content source, downloading images, etc. 

Tracking issue to revert these changes: https://github.com/Azure/azure-sdk-for-python/issues/23381

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_dac_analyze_custom_model.py [hunk 1]
```diff
@@ -18,6 +18,9 @@
 
 class TestDACAnalyzeCustomModel(FormRecognizerTest):
 
+    def teardown(self):
+        self.sleep(4)
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/tests/test_dac_analyze_custom_model.py:22]: 4 seconds keeps the pipeline run around 2 hours, while reducing the frequency of errors.

Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_dac_analyze_custom_model.py:22]: Live pipeline run: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1410278&view=results

**Review by kristapratico — APPROVED** [review]

**Pull request merged**